File: Difficulty: Medium/Topological sort/topological-sort.cpp
class Solution {
  public:
    void dfs(int node,vector<vector<int>>& adjl,vector<int>& vis,stack<int>& st){
        vis[node]=1;
        for(auto i :adjl[node]){
            if(!vis[i]){
                dfs(i,adjl,vis,st);
            }
        }
        st.push(node);
    }
    vector<int> topoSort(int V, vector<vector<int>>& edges) {
        // code here
        vector<vector<int>> adjl(V);
        for(auto edge: edges){
            int u = edge[0];
            int v = edge[1];
            adjl[u].push_back(v);
        }
        stack<int> st;
        vector<int> vis(V,0);
        vector<int> ans;
        for(int i=0;i<V;i++){
            if(!vis[i]){
                dfs(i,adjl,vis,st);
            }
        }
        while(!st.empty()){
            ans.push_back(st.top());
            st.pop();
        }
        return ans;
    }
};

File: Difficulty: Medium/Topological sort/topological-sort.py
class Solution:
    def dfs(self,adj,node,vis,st):
        vis[node]= 1
        for i in adj[node]:
            if not vis[i]:
                self.dfs(adj,i,vis,st)
        
        st.append(node)
    
    def topoSort(self, V, edges):
        # Code here
        adj=[[] for _ in range(V)]
        for u,v in edges:
            adj[u].append(v)
            
            
        st = []
        vis= [False]*V
        
        for i in range(V):
            if not vis[i]:
                self.dfs(adj,i,vis,st)
        
        ans=[]
        while(st):
            ans.append(st.pop())
        return ans            
            